api/app/Models/User.php

```diff
@@ -1041,44 +1039,60 @@ public function scopeAuthorizedToViewSpecific(Builder $query)
                     });
                 });
             }
-            if ($user->isAbleTo('view-own-user')) {
+
+            if ($user?->isAbleTo('view-own-user')) {
                 $query->orWhere('id', $user->id);
             }
         });
+        $filterCountAfter = count($query->getQuery()->wheres);
+        if ($filterCountAfter > $filterCountBefore) {
+            return;
+        }
 
-        return $query;
+        // fall through - query will return nothing
+        $query->where('id', null);
     }
 
-    public function scopeAuthorizedToView(Builder $query)
+    public function scopeAuthorizedToView(Builder $query): void
```

Comment:
I don't remember why we separated this from `scopeAuthorizedToViewSpecific` and I don't see any advantage. Do you? 🤔

Reply:
It is an odd pattern.  I think it's just that we factored out the bulk of the logic and one scope has a quick return for the "any-basicinfo" permission.  Maybe we could find a better name for `scopeAuthorizedToViewSpecific`?
![image](https://github.com/user-attachments/assets/9a8c4fac-6578-4c01-8e53-01c279183581)

Reply:
I see. The difference is that `authorizedToViewSpecific` skips checking `view-any-user` permissions, allowing it to be called by both `authorizedToView` and `authorizedToViewBasicInfo` without any duplication.

However... this still seems pointless and confusing to me now. I suggest ditching `authorizedToViewSpecific`, moving that logic into `authorizedToView`, and just calling `authorizedToView` from `authorizedToViewBasicInfo`. Any objection?

Reply:
Yeah, good call.
[refactor scopeAuthorizedToViewSpecific](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11138/commits/0f123831f9eea3959af35a3859d67b88cbc02f6e)